Sources/Core/Model/Attachment.swift

```
@@ -211,9 +212,78 @@ public extension Attachment {
 }
 
 /// An attachment type.
-public enum AttachmentType: String, Codable {
+public enum AttachmentType: RawRepresentable, Codable, Equatable {
     /// An attachment type.
-    case unknown, image, imgur, giphy, video, youtube, product, file, link
+    case unknown
+    case custom(raw: String)
+    case image
+    case imgur
+    case giphy
+    case video
+    case youtube
+    case product
+    case file
+    case link
+
+    public var rawValue: String? {
+        switch self {
+        case .unknown:
+            return nil
+        case .custom(raw: let raw):
+            return raw
+        case .image:
+            return "image"
+        case .imgur:
+            return "imgur"
+        case .giphy:
+            return "giphy"
+        case .video:
+            return "video"
+        case .youtube:
+            return "youtube"
+        case .product:
+            return "product"
+        case .file:
+            return "file"
+        case .link:
+            return "link"
+        }
+    }
+
+    public init(rawValue: String?) {
+        switch rawValue {
+        case "image":
+            self = .image
+        case "imgur":
+            self = .imgur
+        case "giphy":
+            self = .giphy
+        case "video":
+            self = .video
+        case "youtube":
+            self = .youtube
+        case "product":
+            self = .product
+        case "file":
+            self = .file
+        case "link":
+            self = .link
+        case .some(let raw) where !raw.isEmpty:
+            self = .custom(raw: raw)
+        default:
+            self = .unknown
+        }
+    }
+
+    public init(from decoder: Decoder) throws {
+        let rawValue = try decoder.singleValueContainer().decode(String.self)
+        self = AttachmentType(rawValue: rawValue)
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+        try container.encode(rawValue)
```

Please throw an exception for `unknown` type.